pkg/managed/utils_test.go

```diff
@@ -217,3 +217,51 @@ func TestGetFilteredConnections(t *testing.T) {
 		assert.Contains(expectedClusterNames, name, "connection %s not expected", name)
 	}
 }
+
+func Test_setupHTTPResolve(t *testing.T) {
+	type args struct {
+		url  string
+		host string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "no_url",
+			args:    args{url: "", host: ""},
+			wantErr: false,
+		},
+		{
+			name:    "wrong_url",
+			args:    args{url: ":wrong_url", host: "172.17.0.2"},
+			wantErr: true,
+		},
+		{
+			name:    "no_host",
+			args:    args{url: "https://127.0.0.1", host: ""},
+			wantErr: false,
+		},
+		{
+			name:    "same_host",
+			args:    args{url: "https://127.0.0.1", host: "127.0.0.1"},
+			wantErr: false,
+		},
+		{
+			name:    "different_host",
+			args:    args{url: "https://127.0.0.1", host: "172.17.0.2"},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			os.Setenv("rancherURL", tt.args.url)
+			os.Setenv("rancherHost", tt.args.host)
+			config := &rest.Config{}
+			if err := setupHTTPResolve(config); (err != nil) != tt.wantErr {
```

Comment:
is it possible to assert on the contents of config.Dial?

Reply:
discussed with GZ - this is not possible to test with a simple call to setupHTTPResolve, since it sets up a callback func that only gets called when we send the http request.